examples/de-emphasize.rs

```diff
@@ -0,0 +1,88 @@
+extern crate mdbook;
+extern crate pulldown_cmark;
+extern crate pulldown_cmark_to_cmark;
+
+// This program removes all forms of emphasis from the markdown of the book.
+
+use mdbook::errors::Error;
+use mdbook::MDBook;
+use mdbook::book::{Book, BookItem};
+use mdbook::preprocess::{Preprocessor, PreprocessorContext};
+use pulldown_cmark::{Event, Parser, Tag};
+use pulldown_cmark_to_cmark::fmt::cmark;
+
+use std::ffi::OsString;
+use std::env::{args, args_os};
+use std::process;
+
+struct Deemphasize;
+
+impl Preprocessor for Deemphasize {
+    fn name(&self) -> &str {
+        "md-links-to-html-links"
+    }
+
+    fn run(&self, _ctx: &PreprocessorContext, book: &mut Book) -> ::std::result::Result<(), Error> {
+        eprintln!("Running '{}' preprocessor", self.name());
+        let mut res: Option<_> = None;
+        let mut num_removed_items = 0;
+        book.for_each_mut(|item: &mut BookItem| {
+            if let &Some(Err(_)) = &res {
+                return;
+            }
+            if let BookItem::Chapter(ref mut chapter) = *item {
+                eprintln!("{}: processing chapter '{}'", self.name(), chapter.name);
```

Comment:
I'd probably pull this closure body out into its own function (e.g. `remove_emphasis(ch: &mut Chapter)`). That way the example is easier for people to read and wrap their heads around when they're trying to understand using `mdbook` as a library.

Reply:
I agree! And pulled out a part of it. Thanks to the shared state and the hackery with the result pulling it out on the highest level makes it even more complicated.
However, I think I found a sweet spot that makes it better actually, and I hope you will feel similarly.